### ISSUE TITLE
RCS1096 does not work correctly with composite flags

### ISSUE THREAD

**Issue by Genbox** [dotnet/roslynator#1555]
This seems to have been [reported before](https://github.com/dotnet/roslynator/issues/720), but it is still not working correctly.

Repo:

```csharp
internal class Program
{
    private static FileAttributes _skipMask = FileAttributes.Device | FileAttributes.Offline;

    internal static void Main()
    {
        var fileAttr = FileAttributes.Device;

        //RCS1096 is reported here
        if ((fileAttr & _skipMask) != 0)
            Console.WriteLine("AndCheck: Flags were present");
        else
            Console.WriteLine("AndCheck: Flags not present");

        if (fileAttr.HasFlag(_skipMask))
            Console.WriteLine("HasFlag: Flags were present");
        else
            Console.WriteLine("HasFlag: Flags not present");
    }
}
```

Output:
```
AndCheck: Flags were present
HasFlag: Flags not present
```
